docs/api/security/psa.md

```diff
@@ -0,0 +1,38 @@
+## Mbed PSA
+
+### Terms and Abbreviations
+
+| Abbreviation | Description                         |
+|--------------|-------------------------------------|
+| PSA          | Platform Security Architecture      |
+| SPM          | Secure Partition Manager            |
+| SPE          | Secure Processing Environment       |
+| NSPE         | Non-Secure Processing Environment   |
+| IPC          | Inter Process Communication         |
+| RoT          | Root Of Trust                       |
+
+
+### Overview
+Mbed PSA provides essential root of trust services and infrastructure for developing robust IoT applications.
+When mbed-os is running on PSA Security Model compliant target, Mbed PSA helps to protect cryptographic assets, credentials, and critical code sections by providing an isolation between a Secure Processing Environment (SPE) and a Non-Secure Processing Environment (NSPE). The isolation is managed by Secure Partition Manager (SPM) which utilizes unique HW features available on the target. SPM provides standardized IPC APIs which abstract the fact that partitions could be living inside a virtualized environment (v8M, TEE on Cortex-A), or another chip.
+
+Mbed PSA bridges the differences between PSA and Non-PSA targets for application developers, allowing to use same standard PSA APIs on both target types.
+Mbed PSA provides PSA API compliance for developing robust IoT applications and 
+allows to choose target type at later phase according to final application threat model.
+
+### Platform types
+Mbed PSA supports following platform types:
+- Non PSA platform - these are single core ARMv7-M targets. On these targets
+  Mbed PSA provides same PSA services exposing PSA APIs as it would on PSA targets. 
+  PSA emulation layer allows seamless software portability to more 
+  security oriented targets.
+- Asymmetric Multiprocessing (AMP) systems - Multi core ARMv7-M targets (e.g PSoC6
+  featuring CM4 and CM0+ cores). On these targets one of the cores is dedicated to PSA usage only and implements SPE. Mbed PSA provides PSA APIs proxy implementation on non-secure core, which redirect execution to SPE.
```

Comment:
"On these targets one of the cores is dedicated to PSA usage only and implements SPE" - This is not correct.
It is SPM that is running on one target only. PSA, which is not a module but more of a model consists of several specifications, is not running anywhere :)